src/lib/executeSourceTransfer.js

```diff
@@ -23,19 +29,44 @@ function * executeSourceTransfer (destinationTransfer, fulfillment, ledgers, bac
   // TODO check the timestamp on the response from the ledger against
   // the transfer's expiry date
   // See https://github.com/interledgerjs/five-bells-ledger/issues/149
-  yield ledgers.getPlugin(sourceTransferLedger)
-    .fulfillCondition(sourceTransferID, fulfillment)
-    .then(() =>
-      backend.submitPayment({
-        source_ledger: sourceTransferLedger,
-        source_amount: sourceTransferAmount,
-        destination_ledger: destinationTransfer.ledger,
-        destination_amount: destinationTransfer.amount
+  const plugin = ledgers.getPlugin(sourceTransferLedger)
+
+  yield promiseRetry(retryOpts, function (retry, number) {
+    return Promise.resolve(plugin.fulfillCondition(sourceTransferID, fulfillment))
+      .catch(function (err) {
+        if (shouldRetry(err)) {
+          log.debug(`Resubmitting fulfillment for source transfer ${sourceTransferID}.`)
+          retry(err)
+        }
       })
-    )
-    .catch(() => {
-      log.error('Attempted to execute source transfer but it was unsucessful: we have not been fully repaid')
+  })
+  .then(function () {
+    backend.submitPayment({
+      source_ledger: sourceTransferLedger,
+      source_amount: sourceTransferAmount,
+      destination_ledger: destinationTransfer.ledger,
+      destination_amount: destinationTransfer.amount
     })
+  })
+  .catch(() => {
+    log.error('Attempted to execute source transfer but it was unsucessful: we have not been fully repaid.')
+  })
+}
+
+function shouldRetry (err) {
+  // Refer to error types defined in the specs:
+  // https://github.com/interledger/rfcs/blob/master/0004-ledger-plugin-interface/0004-ledger-plugin-interface.md#errors
+
+  // returns true for UnreachableError and any other error not listed below
+  return (err.name !== 'InvalidFieldsError' ||
+          err.name !== 'TransferNotFound' ||
+          err.name !== 'MissingFulfillmentError' ||
+          err.name !== 'DuplicateIdError' ||
```

Comment:
It shouldn't catch MissingFulfillment or DuplicateId errors because those don't make any sense in this context. If you get one of those there's something wrong with the plugin